generative-ai/zero-shot-classification.ipynb

```diff
@@ -0,0 +1,271 @@
+    "\n",
+    "### What models are best for zero-shot classification?\n",
+    "\n",
+    "Choosing the right model for your zero-shot classification depends on your classification task. It’s no surprise that there is an abundance to choose from, let’s consider three types of models:  \n",
+    "  \n",
+    "- **Zero-shot classification model** is a type of model that can classify data into categories without task-specific labeled examples during the prediction phase. The models rely on training data, which is normally a large-scale, general dataset to classify new, unseen classes. One of the most popular zero-shot classification models for zero-shot text classification is HuggingFace’s facebook/[bart-large-mnli](https://huggingface.co/tasks/zero-shot-classification) model based on the BART-large transformer model architecture. Zero-shot classification models perform well on generalized tasks but, because there is no fine-tuning on specific tasks or datasets, accuracy may be limited. Because of this limitation, the model requires well-formulated prompts.\n",
+    "\n",
+    "- **Large Language Models** (LLMs) like GPT and BERT are designed to perform a variety of natural language processing tasks. This can limit use cases using multimodal datasets such as images and audio. These models are designed to handle text data, often using deep learning architectures likes [transformers](https://www.ibm.com/topics/transformer-model). LLMs are trained on a large corpora of text data, giving them extensive knowledge of language, syntax, semantics, and some domain-specific knowledge. These models generally perform well with little to no task-specific fine-tuning, making them suitable for zero-shot and few-shot classification scenarios. Due to their generalized training, LLMs may have limited accuracy for specialized tasks especially ones that require domain specific data. These models are best to work with when the dataset is text-based.\n",
+    "\n",
+    "- **Foundation models** are multimodal, meaning they are often trained on a myriad of modalities including text, images, and speech. These models are generally versatile and after pretraining can undergo optimization for many different tasks. IBM’s granite models classify data by using a large language model (LLM) trained on a curated dataset of business-relevant information, including legal, financial, and technical domains, which allows it to analyze text and identify patterns to categorize data into specific classes based on the context and semantic meaning within the text. Because of their multimodal capacity, these types of models can handle image and text classification. These models are ideal when you need a broad range of capabilities or want to handle multiple types of data for instance, image or audio classification.\n",
```

Comment:
I think this needs to be upper case Granite Model and a TM

Reply:
Updated the capitalization and added (TM) to this and one other instance where "IBM Granite Model" is used